enrolments/validators.py

```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
+from django.core.exceptions import ValidationError
+from registration.models import Student
+
+
+def validate_attendance(classObj):
+    """
+    Validates against the attendance column for the Class model.
+    The expected structure is:
+    [
+        {
+            "date": "2021-04-19",
+            "attendees": [
+                // student IDs
+                1, 2, 3, 4, 5
+            ]
+        },
+        {
+            "date": "2021-04-26",
+            "attendees": [
+                // student IDs
+                1, 2, 3, 4, 5
+            ]
+        }
+    ]
+    """
+    for session in classObj:
+        date = session["date"]
+        attendees = session["attendees"]
+        try:
+            datetime.strptime(date, "%Y-%m-%d").date()
+        except ValueError:
+            raise ValidationError(
+                "%(date) must be formatted as YYYY-MM-DD and must be a valid date",
+                code="invalid",
+                params={"date": date},
+            )
+        if attendees and not Student.objects.filter(pk__in=attendees).exists():
+            raise ValidationError(
+                "one or more %(attendee)(s) do not exist",
+                code="invalid",
```

Comment:
ahh I see what you mean by only the generic "Enter a valid JSON." message showing now

I think the "invalid" code is protected somehow? it seems renaming `code="invalid"` to something like `code="invalid_attendee"` gets your custom message to show!

<img width="678" alt="Screen Shot 2021-03-01 at 1 22 34 AM" src="https://user-images.githubusercontent.com/37346399/109459349-9d3b7000-7a2c-11eb-88e6-c9720f4a1f3a.png">